File: 0053-maximum-subarray/0053-maximum-subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        int currSum = nums[0];
        int maxSum = nums[0];

        for(int i = 1; i<nums.Length; i++)
        {
            currSum = currSum + nums[i] < nums[i]? nums[i]: currSum+nums[i];
            maxSum = Math.Max(currSum, maxSum);
        }
        return maxSum;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if(nums.size()==1) return nums[0];
        int n = nums.size();
        int sum = nums[0];
        int maxSum = nums[0];
        
        for(int i=1;i<n;i++){
            sum = max(nums[i], sum+nums[i]);
            maxSum = max(maxSum, sum);
        }
        return maxSum;
    }
};